### DISCUSSION TITLE
How to use defineLoader without file-based routes

### DISCUSSION THREAD

**Discussion by blacksheep107** [posva/unplugin-vue-router#456] [Questions and Help] [answered]
v0.10.1
The loader havn't run when the route params changes; I'm using origin vue-router's config, not migrating as https://uvr.esm.is/introduction.html#migrating-an-existing-project

Can I use defineLoader properly without migration?

**Comment by posva** [comment, marked as answer]
Yes, it should work when installed: https://uvr.esm.is/data-loaders/
